spiketoolkit/validation/quality_metric_classes/noise_overlap.py

```diff
@@ -23,47 +26,75 @@ def __init__(self, metric_data):
         if not metric_data.has_recording():
             raise ValueError("MetricData object must have a recording")
 
-    def compute_metric(self, max_spikes_per_unit_for_noise_overlap, num_features, num_knn, **kwargs):
+    def compute_metric(self, num_channels_to_compare, max_spikes_per_unit_for_noise_overlap,
+                        num_features, num_knn, **kwargs):
+
```

Comment:
Since this implementation relies on `max_spikes_per_unit_for_noise_overlap` not being `None`, I would add:
```
assert max_spikes_per_unit_for_noise_overlap is not None, "'max_spikes_per_unit_for_noise_overlap' must be an integer."
```

Reply:
Added this